examples/run/common/throughput_mt.ipp

```diff
@@ -80,17 +80,17 @@ int throughput_mt(std::string_view description, int argc, char* argv[],
     HOST_MR uncached_host_mr;
 
     // Read in all input events into memory.
-    demonstrator_input input;
+    demonstrator_input input(&uncached_host_mr);
+
     {
         performance::timer t{"File reading", times};
-        for (unsigned int event = 0; event < throughput_cfg.loaded_events;
-             ++event) {
-            input = io::read(
-                throughput_cfg.loaded_events, throughput_cfg.input_directory,
-                throughput_cfg.detector_file,
-                throughput_cfg.digitization_config_file,
-                throughput_cfg.input_data_format, &uncached_host_mr);
+        for (std::size_t i = 0; i < throughput_cfg.loaded_events; ++i) {
+            input.push_back(&uncached_host_mr);
```

Comment:
Let's be a bit more explicit with this code. I know what is happening here, but to a lay reader it would be pretty confusing. (Since an auto-conversion is taking place here.)